src/sources/vector/v1.rs

```diff
@@ -82,12 +82,16 @@ impl GenerateConfig for VectorConfig {
 impl VectorConfig {
     pub(super) async fn build(&self, cx: SourceContext) -> crate::Result<Source> {
         let vector = VectorSource;
-        let tls = MaybeTlsSettings::from_config(&self.tls, true)?;
+        let tls_config = self.tls.as_ref().map(|tls| tls.tls_config.clone());
+        let tls_client_metadata_key = self.tls.as_ref().and_then(|tls| tls.client_metadata_key.clone());
+
+        let tls = MaybeTlsSettings::from_config(&tls_config, true)?;
```

Comment:
This pattern appears to be found in most of the modified sources here (at least four times).
Would it be feasible and advantageous to extract this into a helper function say in `sources/tls/mod.rs` , which returns something like `crate::Result<(tls, tls_client_metadata_key)>` ?

Reply:
Doesn't sound like a bad idea, but I also don't love holding up this PR any more than necessary - it's been dragging for a while. I defer to @jszwedko

Reply:
Just a minor suggestion, no need to hold the PR just for this comment.